File: 121 - Best Time to Buy and Sell Stock/array.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max = 0
        min = prices[0]

        for price in prices:
            if price < min:
                min = price

            if (price - min) > max:
                max = (price - min)
        
        return max
        

File: 121 - Best Time to Buy and Sell Stock/array.go
func maxProfit(prices []int) int {
	max := 0
	min := prices[0]

	for i := 0; i < len(prices); i++ {
		if min > prices[i] {
			min = prices[i]
		}
		if (prices[i] - min) > max {
			max = (prices[i] - min)
		}
	}

	return max
}